### ISSUE TITLE
word dropout or word embedding dropout?

### ISSUE THREAD

**Issue by mengxuehu** [timbmg/Sentence-VAE#5]
According to the papar, "We do this by randomly **replacing** some fraction of the conditioned-on **word tokens** with the generic unknown word token unk", but it seems that [this implementation](https://github.com/timbmg/Sentence-VAE/blob/e1df0eb3469defb56adbd9a7d3babe0557cd5a14/model.py#L85) is dropping out word embedding.

**Comment by nick11roberts** [comment]
According to [this line](https://github.com/timbmg/Sentence-VAE/blob/e1df0eb3469defb56adbd9a7d3babe0557cd5a14/model.py#L55), dropout is being applied to the sequence of embeddings after they have been looked up in the embedding. 

I'm not sure if this is right, because I think that this has the effect of randomly dropping elements of the embedding vector corresponding to a token rather than dropping whole embedding vectors (post lookup). Perhaps as opposed to using `torch.nn.Dropout`, `torch.nn.Dropout2d` should be applied along the sequence dimension. Does this seem reasonable?

**Comment by timbmg** [comment]
Right, this is different from the paper. Unfortunately, I don't have the time to fix this right now. But feel free to do a PR. If you do, please also keep the dropout version as an option.

**Comment by nguyenvo09** [comment]
How about testing phase? We do need to disable dropout right? Based on current implementation, dropout still works in the testing phase.

**Comment by timbmg** [comment]
This does that:
‘’’python
if split == 'train':
                model.train()
            else:
                model.eval()
‘’’
